picoshare/guest_link.go

```diff
@@ -9,22 +9,24 @@ type (
 	GuestLinkLabel          string
 	GuestUploadMaxFileBytes *uint64
 	GuestUploadCountLimit   *int
-
-	GuestLink struct {
+	GuestUploadDisabled     *int
```

Comment:
Custom types are useful when we want to restrict what value can be in the native type. For example, a `GuestLinkID` is different than a `string` because not all valid `string`s are valid `GuestLinkID`s, so we have a type to restrict the values.

With a boolean, there's not much value in a custom type, so I think we should just add a boolean type to the `GuestLink` struct.

SQLite has to store it as an integer because it doesn't have a boolean type, but everything outside of the sqlite package should treat the value as a `bool`, and our `store/sqlite` package should be responsible for serializing/deserializing it to an integer for storage.